staging/src/k8s.io/client-go/tools/clientcmd/config.go

```diff
@@ -159,6 +159,9 @@ func ModifyConfig(configAccess ConfigAccess, newConfig clientcmdapi.Config, rela
 	// to avoid deadlock (note: this can fail w/ symlinks, but... come on).
 	sort.Strings(possibleSources)
 	for _, filename := range possibleSources {
+		if _, err := os.Stat(lockName(filename)); err == nil {
```

Comment:
This defeats the purpose of the lock file by tolerating its existence from other processes.

Reply:
That's true, thanks!
I'm gonna close this pr and reference #53880 to that issue.